### PR TITLE
Add coredomain and service parent pom latest version check rules

### PR DESCRIPTION
**Pull request by santhosh-hmcts**

### CODE REVIEW

**Review by arcadiushmcts — APPROVED** [review]
Looks good to me, @allanmckenzie could you also have a look please

File: .travis.yml [hunk 1]
```diff
@@ -2,7 +2,7 @@ dist: trusty
 sudo: required
 language: java
```
Comment (allanmckenzie) [inline .travis.yml:2]: I'm not sure that all these are correct. The correct versions of these properteis shoud be

dist: xenial
language: java
os: linux
jdk:
  - openjdk17

File: src/test/java/uk/gov/justice/maven/rules/RequireLatestDependencyVersionRuleTest.java [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+    @Mock
+    private DependencyVersionCheckingService dependencyVersionCheckingService;
+
+    @Mock
+    private ArtifactFinderFactory artifactFinderFactory;
+
+    @InjectMocks
+    private RequireLatestDependencyVersionRule requireLatestDependencyVersionRule;
+
+    @BeforeEach
```
Comment (allanmckenzie) [inline src/test/java/uk/gov/justice/maven/rules/RequireLatestDependencyVersionRuleTest.java:41]: I tend to think that @BeforeEach to create your objects or to do verification is an anti-pattern. It assumes that all tests both now in in the future will behave in exactly the same way. Often they change over time. I think it's better to do this in the tests themselves. It makes the tests easier to read and allows for future changes

File: src/test/java/uk/gov/justice/maven/rules/RequireLatestDependencyVersionRuleTest.java [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+
+    @Nested
+    class ExecuteTest {
+
+        private MavenProject mavenProject;
+        private Log log;
+        private ArtifactFinder artifactFinder;
+        private EnforcerRuleHelper helper;
+
+        @BeforeEach
```
Comment (allanmckenzie) [inline src/test/java/uk/gov/justice/maven/rules/RequireLatestDependencyVersionRuleTest.java:55]: Again - avoid @BeforeEach if you possibly can

Comment (santhosh-hmcts) [reply src/test/java/uk/gov/justice/maven/rules/RequireLatestDependencyVersionRuleTest.java:55]: Will address both

**Review by allanmckenzie — APPROVED** [review]

**Pull request merged**